### PR TITLE
Initial commit for OFMCC-1594: Add inclusion policy yes/ and doc uplo…

### PR DESCRIPTION
**Pull request by jgstorey**
…ad to organization.

### CODE REVIEW

File: frontend/src/components/organizations/OrganizationInfo.vue [hunk 1]
```diff
@@ -221,50 +224,146 @@ export default {
...
     },
 
+    setLoadingState(isLoading) {
+      this.localLoading = isLoading;
+      this.$emit('update:loading', isLoading);
+    },
+
+    async refreshDocuments() {
+      const documents = await DocumentService.getDocuments(this.organization.organizationId);
+      this.initializeDocuments(documents);
```
Comment (weskubo-cgi) [inline frontend/src/components/organizations/OrganizationInfo.vue:301]: This same code is used on line 257 but without a method call. We should be consistent. I don't think the method is necessary given that it an be done in one line and it calls yet another custom method.

File: frontend/src/components/organizations/OrganizationInfo.vue [hunk 2]
```diff
@@ -221,50 +224,146 @@ export default {
...
+    this.loadData()
   },
   methods: {
-    async update() {
-      this.$refs.organizationForm?.validate()
-      if (!this.isFormComplete) {
+    async loadData() {
+      try {
+        this.localLoading = true
+        this.organizationEdit = { ...this.organization, hasInclusionPolicy: this.organization.hasInclusionPolicy || false }
```
Comment (weskubo-cgi) [inline frontend/src/components/organizations/OrganizationInfo.vue:265]: I assume this is because the org is created with a null value instead of false?

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/components/organizations/OrganizationInfo.vue [hunk 3]
```diff
@@ -221,50 +224,146 @@ export default {
...
-  computed: {
-    ...mapState(useAuthStore, ['userInfo']),
+  watch: {
+    documentsToUpload: {
+      handler(val) {
+        if (val.length === 0 && this.uploadedDocuments.length === 0 && !this.isLoading) {
+          this.showUploadDocumentsAlert = true
+        } else {
+          this.showUploadDocumentsAlert = false
+        }
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/organizations/OrganizationInfo.vue:247]: Fixed.

File: frontend/src/components/organizations/OrganizationInfo.vue [hunk 4]
```diff
@@ -221,50 +224,146 @@ export default {
...
+      uploadedDocumentsInitialValue: [],
+      documentsToUpload: [],
+      documentsToDelete: [],
+      showUploadDocumentsAlert: false,
     }
   },
-  computed: {
-    ...mapState(useAuthStore, ['userInfo']),
+  watch: {
+    documentsToUpload: {
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/organizations/OrganizationInfo.vue:241]: Not required, i used the watch as an easy way to suppress the message if the user invoked 'add file' or 'select a file'... so the message would not persist if they were following instructions. I removed so message will persist until save and persist if they invoke save and no files selected to upload or uploaded.

File: frontend/src/components/organizations/OrganizationInfo.vue [hunk 5]
```diff
@@ -221,50 +224,146 @@ export default {
...
+      handler(val) {
+        if (val.length === 0 && this.uploadedDocuments.length === 0 && !this.isLoading) {
+          this.showUploadDocumentsAlert = true
+        } else {
+          this.showUploadDocumentsAlert = false
+        }
+      },
+    },
+    'organizationEdit.hasInclusionPolicy': {
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/organizations/OrganizationInfo.vue:250]: Fixed.

File: frontend/src/components/organizations/OrganizationInfo.vue [hunk 6]
```diff
@@ -221,50 +224,146 @@ export default {
       default: false,
     },
   },
+  emits: ['update:loading'],
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/organizations/OrganizationInfo.vue:227]: Updated to componentLoading

File: frontend/src/views/account-mgmt/ManageOrganizationView.vue [hunk 7]
```diff
@@ -147,7 +148,21 @@ export default {
...
      */
     openChangeRequestDialog() {
-      this.setWarningAlert('This feature is not yet implemented')
+      this.setWarningAlert('This feature will be implemented in a future sprint')
+    },
+
+    /**
+     * Set the loading state of the component
+     */
+    setLoadingComponent(value) {
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/views/account-mgmt/ManageOrganizationView.vue:157]: Updated to suggested.

File: frontend/src/components/organizations/OrganizationInfo.vue (hunk 1 above)
Comment (jgstorey) [reply frontend/src/components/organizations/OrganizationInfo.vue:301]: You're right, not necessary... i just thought it made the readability better in the save code. Updated.

File: frontend/src/components/organizations/OrganizationInfo.vue (hunk 2 above)
Comment (jgstorey) [reply frontend/src/components/organizations/OrganizationInfo.vue:265]: Correct.

File: frontend/src/components/organizations/OrganizationInfo.vue [hunk 8]
```diff
@@ -221,50 +224,146 @@ export default {
...
+      if (this.$refs.documentUpload) {
+        this.$refs.documentUpload.resetDocuments()
+      }
     },
 
     isLoading() {
       return this.loading || this.localLoading
     },
 
+    updateDocumentsToUpload({ documents, areValidFilesUploaded }) {
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/organizations/OrganizationInfo.vue:317]: Fixed.

**Pull request closed without merging**